cilium: Add workflows for GKE in tunnel mode, with and without encryption

jrfastab opened the pull request:
We want to test vxlan with encryption add workflow tests to GKE to capture this use case.

jrfastab committed:
cilium: Add GKE workflow to test vxlan

Add vxlan test path for `./cilium connectivity test` at the moment
we do not cover this path with workflow CI.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

jrfastab committed:
cilium: Add vxlan datapath encryption tests in GKE

We do not have a connectivity test for encryption with tunneling this
adds on on GKE.

We also bump the timer slightly to account for the extra runs.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

nbusseneau commented on .github/workflows/gke.yaml:
```diff
@@ -160,6 +199,45 @@ jobs:
+      - name: Install Cilium With Encryption and Tunnel Datapath
+        run: |
+          cilium install \
+            --cluster-name=${{ env.clusterName }} \
```
I think we're missing something here... 😇 
```suggestion
            --encryption \
            --cluster-name=${{ env.clusterName }} \
```

jrfastab replied:
oops good catch I've add this now.

nbusseneau reviewed (commented):
[[Link to Cilium Slack thread to discuss this]](https://cilium.slack.com/archives/C7PE7V806/p1618406776180900)

To be honest I am a bit concerned... At first we had a simple, straightforward workflow as in "just run `cilium install` and then `cilium connectivity test`, boom", with some fluff around for artifacts etc.

Then we added another set for encryption. Now we're adding two more for tunnel datapath mode.

My concern is: when does it end? 🤔 
- Is it really what we want to do and move forward with regarding platform testing via CI 3.0?
- Would it make sense to have separate workflows for separate concerns, rather than bundling four (!) `cilium install` / `cilium connectivity test` in a single workflow?
- Would it make sense to ditch the current `cilium install` / `cilium connectivity test` approach and return to the drawing board so that a single `cilium install` / `cilium connectivity test` (perhaps with special CLI parameters) does them all?

If we keep the current approach and do want to bundle these in a single workflow, then I will make a note of doing a refactoring pass in order to reduce the amount of duplication in here.

Disclaimer: I'm not a datapath guy, so I have literally zero idea what makes sense or not.

jrfastab committed:
cilium: gke workflow noticed we missed initial syn/ack bump sleep

We missed the initial syn/ack from pod-to-pod test,

❌ SYN-ACK and(ip(src=10.16.1.24,dst=10.16.1.186),tcp(srcPort=8080),tcpflags(syn,ack)) not found for pod cilium-test/client-6c46bff775-ls4qm

however the command did succeed so it must have been there, but missed.
Lets bump the sleep slightly to be sure hubble and port forwarding is
running. Changed from 15minutes to 30 mins.

Also changing restart-pods to true seems to make tests stable.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>

nbusseneau commented on .github/workflows/gke.yaml:
```diff
@@ -97,7 +97,41 @@ jobs:
             --cluster-name=${{ env.clusterName }} \
             --wait=false \
-            --restart-unmanaged-pods=false \
+            --restart-unmanaged-pods=true \
```
I think we can actually just drop all the `--restart-unmanaged-pods=true` flags since that's the default anyway.

jrfastab replied:
ah yeah lets do it.
